File: SYP.Web/Modules/ServerTypes/Products/BrandsEditorColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { BrandsRow } from "./BrandsRow";

export interface BrandsEditorColumns {
    Id: Column<BrandsRow>;
    Name: Column<BrandsRow>;
    Description: Column<BrandsRow>;
    Logo: Column<BrandsRow>;
    IsActive: Column<BrandsRow>;
    InsertDate: Column<BrandsRow>;
    InsertUserId: Column<BrandsRow>;
    UpdateDate: Column<BrandsRow>;
    UpdateUserId: Column<BrandsRow>;
}

export class BrandsEditorColumns extends ColumnsBase<BrandsRow> {
    static readonly columnsKey = 'Products.BrandsEditor';
    static readonly Fields = fieldsProxy<BrandsEditorColumns>();
}

File: SYP.Web/Modules/Products/Brands/Editor/BrandsEditorColumns.cs
namespace SYP.Products.Columns;

[ColumnsScript("Products.BrandsEditor")]
[BasedOnRow(typeof(BrandsRow), CheckNames = true)]
public class BrandsEditorColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Description { get; set; }
    public string Logo { get; set; }
    public bool IsActive { get; set; }
    public DateTime InsertDate { get; set; }
    public int InsertUserId { get; set; }
    public DateTime UpdateDate { get; set; }
    public int UpdateUserId { get; set; }
}